test/examples/uct_hello_world.c

```diff
@@ -1,17 +1,635 @@
+    uct_tl_resource_desc_t  *tl_resources; /*Communication resource descriptor */
+    unsigned                num_md_resources; /* Number of protected domain */
+    unsigned                num_tl_resources; /* Number of transport resources resource objects created */
+    uct_md_config_t         *md_config;
+    ucs_status_t            status;
+    int                     i;
+    int                     j;
+
+    status = uct_query_md_resources(&md_resources, &num_md_resources);
+    CHKERR_JUMP(UCS_OK != status, "query for protected domain resources", error_ret);
+
+    /* Iterate through protected domain resources */
+    for (i = 0; i < num_md_resources; ++i) {
+        status = uct_md_config_read(md_resources[i].md_name, NULL, NULL, &md_config);
+        CHKERR_JUMP(UCS_OK != status, "read PD config", release_pd);
+
+        status = uct_md_open(md_resources[i].md_name, md_config, &iface_p->pd);
+        uct_config_release(md_config);
+        CHKERR_JUMP(UCS_OK != status, "open protected domains", release_pd);
+
+        status = uct_md_query_tl_resources(iface_p->pd, &tl_resources, &num_tl_resources);
+        CHKERR_JUMP(UCS_OK != status, "query transport resources", close_pd);
+
+        /* Go through each available transport and find the proper name */
+        for (j = 0; j < num_tl_resources; ++j) {
+            if (!strcmp(cmd_args->dev_name, tl_resources[j].dev_name) &&
+                !strcmp(cmd_args->tl_name, tl_resources[j].tl_name)) {
+                status = init_iface(tl_resources[j].dev_name,
+                                    tl_resources[j].tl_name,
+                                    cmd_args->func_am_type, iface_p);
+                if (UCS_OK == status) {
+                    fprintf(stdout, "Using %s with %s.\n",
+                            tl_resources[j].dev_name,
+                            tl_resources[j].tl_name);
+                    fflush(stdout);
+                    uct_release_tl_resource_list(tl_resources);
+                    goto release_pd;
+                }
+            }
+        }
+        uct_release_tl_resource_list(tl_resources);
+        uct_md_close(iface_p->pd);
+    }
+
+    fprintf(stderr, "No supported (dev/tl) found (%s/%s)\n",
+            cmd_args->dev_name, cmd_args->tl_name);
+    status = UCS_ERR_UNSUPPORTED;
+
+release_pd:
+    uct_release_md_resource_list(md_resources);
+error_ret:
+    return status;
+close_pd:
+    uct_md_close(iface_p->pd);
+    goto release_pd;
+}
+
+int print_err_usage()
+{
+    fprintf(stderr, "Usage: uct_hello_world [parameters]\n");
+    fprintf(stderr, "UCT hello world client/server example utility\n");
+    fprintf(stderr, "\nParameters are:\n");
+    fprintf(stderr, "  -i      Select \"uct_ep_am_short\" function to "
+            "send the message (default)\n");
+    fprintf(stderr, "  -b      Select \"uct_ep_am_bcopy\" function to "
+            "send the message\n");
+    fprintf(stderr, "  -z      Select \"uct_ep_am_zcopy\" function to "
```

Comment:
It might be better to use func_am_t_str() here to avoid function name string duplication.